### ISSUE TITLE
Restrict fair workflows to only consist of is_fairstep-decorated functions

### ISSUE THREAD

**Issue by svenvanderburg** [fair-workflows/fairworkflows#150]
Defining a fair workflow using `@is_fairworkflow` decorator should throw an error if there are functions in it that are not decorated with `@is_fairstep`

**Comment by svenvanderburg** [comment]
In #159 we fixed one case, but this one still fails with a user-unfriendly error message:
```python 

    def test_workflow_mixed_decorated_steps(self):
        def add(a: float, b: float) -> float:
            """Adding up numbers. NB: no is_fairstep decorator!"""
            return a + b

        @is_fairstep(label='Subtraction')
        def sub(a: float, b: float) -> float:
            """Subtracting numbers."""
            return a - b

        @is_fairworkflow(label='My Workflow')
        def my_workflow(in1, in2):
            """
            A simple addition, subtraction workflow
            """
            return add(in1, sub(in2, in2))

        fw = my_workflow(1, 2)
        assert isinstance(fw, FairWorkflow)
```
this should be easier to fix after #160

**Comment by svenvanderburg** [comment]
In #167 the solution introduced in #159 actually catches even less wrong usage of the decorators. I think we should catch all TypeErrors, and reraise with the original error message + pointing the user to the fact that that should use the decorators.
